src/components/task/task-slice.ts

```diff
@@ -0,0 +1,183 @@
+import {createSelector, createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {RootState} from "../../app/store";
+import dayjs, {Dayjs} from 'dayjs';
+import { v4 as uuidv4 } from 'uuid';
+
+export interface Task {
+    id: string;
+    name: string;
+    description: string;
+    completed: boolean;
+    scheduledDate: string;
+    // createdAt: string;
+    // dayOfWeek: number;
+}
+
+interface TaskState {
+    name: string;
+    description: string;
+    tempScheduledDate: string;
+    currentStep: number;
+    tasks: Task[];
+    completedTasks: Task[];
```

Comment:
I think you may simplify it a bit in a future. Instead of storing two different arrays for "tasks" and "completedTasks" you can go with a single array of tasks and just add "completed" property to the Task interface.

Reply:
Oh, you actually have this property in Taks interface. Then you don't even need an extra array with competed tasks